Library/OcCryptoLib/RsaDigitalSign.c

```diff
@@ -0,0 +1,511 @@
+/** @file
+
+OcCryptoLib
+
+Copyright (c) 2018, savvas
+
+All rights reserved.
+
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+/**
+  Copyright (c) 2014 The Chromium OS Authors. All rights reserved.
+  Use of this source code is governed by a BSD-style license that can be
+  found in the LICENSE file.
+
+  Implementation of RSA signature verification which uses a pre-processed key
+  for computation.
+**/
+
+#ifdef EFIAPI
+#include <Library/BaseMemoryLib.h>
+#endif
+
+#include <Library/DebugLib.h>
+#include <Library/BaseLib.h>
+#include <Library/OcGuardLib.h>
+#include <Library/OcCryptoLib.h>
+
+/**
+  PKCS#1 padding (from the RSA PKCS#1 v2.1 standard)
+
+  The DER-encoded padding is defined as follows :
+  0x00 || 0x01 || PS || 0x00 || T
+
+  T: DER Encoded DigestInfo value which depends on the hash function used,
+  for SHA-256:
+  (0x)30 31 30 0d 06 09 60 86 48 01 65 03 04 02 01 05 00 04 20 || H.
+
+  Length(T) = 51 octets for SHA-256
+
+  PS: octet string consisting of {Length(RSA Key) - Length(T) - 3} 0xFF
+ **/
+#define PKCS_PAD_SIZE (CONFIG_RSA_KEY_SIZE - SHA256_DIGEST_SIZE)
+
+STATIC  UINT8 mSha256Tail[] = {
+  0x00, 0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60,
+  0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01,
+  0x05, 0x00, 0x04, 0x20
+};
+
+UINT64
+Mula32 (
+  UINT32  A,
+  UINT32  B,
```

Comment:
Please add IN/OUT annotations to all functions, as currently it is a bit hard to figure out data transfers.